### PR TITLE
LPD-32949 & LPD-26404 Permissions to be asked during publication and not when autosaving

### PR DESCRIPTION
**Pull request by DiegoHu97**
LPD-32949 & LPD-26404 Permissions to be asked during publication and not when autosaving

# What is this trying to solve?

We want to set the article's permissions the first time we publish the article. So the first time we publish an article we should show the permissions modal.

[Link to ticket](https://liferay.atlassian.net/browse/LPD-32949)
[Link to ticket](https://liferay.atlassian.net/browse/LPD-26404)

# How am I fixing it?

- We have added a property in **"JournalEditArticleDisplayContext.java"** called `showPublishModal` that controlles whether the article has been published or it's in draft.
- Then, from the frontend part, we have adapted the `showPublishModal` condition in order to show the modal the first time we publish the article. 
- Finally, this change needed a change in **"JournalArticleLocalServiceImpl.java** to add the permissions selected when updating the article, as with the autosave the `articleId` will be never null.

# How can I test it?

- Enable FF 'LPD-11228': true,
		'LPD-15596': true,
- Create a WC
- Fill the title 
- Wait for the Autosave indicator
- Click on Publish button
- Select Publish with Permissions
- Select some Permissions
- Publish the article
- Click on the article's kebab > Permissions
- See the permissions selected are enabled

https://github.com/user-attachments/assets/80de53ae-6485-421c-91e6-d2ba8735c6af

### CODE REVIEW

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:sf

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/modals/PublishModal.js [hunk 1]
```diff
@@ -69,14 +69,14 @@ export default function PublishModal({
...
-				{articleId ? null : (
+				{!articleId || Liferay.FeatureFlags['LPD-11228'] ? (
 					<div className="mt-3">
 						<PermissionsOptions
 							formId={formId}
 							permissionsURL={permissionsURL}
 						/>
 					</div>
-				)}
+				) : null}
```
Comment (boton) [inline modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/modals/PublishModal.js:79]: Now the conditional is a real boolean we can delete the `:null` part

```jsx
{(!articleId || Liferay.FeatureFlags['LPD-11228']) && (
	<div className="mt-3">
		<PermissionsOptions
			formId={formId}
			permissionsURL={permissionsURL}
		/>
	</div>
)}
```

**Comment by boton** [comment]
ci:test:playwright

**Comment by Mikellorza** [comment]
LGTM!!  Great job Diego!!

**Review by Mikellorza — APPROVED** [review]

**Comment by DiegoHu97** [comment]
Hey @boton , I've just made the change, the failing tests seems not to be related with this PR, thanks for the review! :smile:

**Comment by DiegoHu97** [comment]
ci:test:sf

**Review by Mikellorza — APPROVED** [review]

**Comment by Mikellorza** [comment]
ci:forward

**Comment by DiegoHu97** [comment]
ci;report:54313081

**Comment by DiegoHu97** [comment]
ci:forward

**Comment by DiegoHu97** [comment]
The failing test is not related

**Comment by DiegoHu97** [comment]
ci:forward:force

**Pull request closed without merging**